03/main.c

```diff
@@ -0,0 +1,139 @@
+/*
+* Homework 03
+* By Ilya Nikolskiy
+* 28.09.2020
+*1. Используя заголовочный файл <math.h>, описать функцию: int calcSquareEq(int a, int b, int c, double* x1, double* x2); Которая будет решать квадратное уравнение вида a * x ** 2 + b * x + c = 0, и записывать корни этого уравнения в переменные, адреса которых переданы в качестве указателей х1 и х2. Функция должна вернуть -1, если уравнение не имеет корней, 0, если у уравнения есть один корень, и 1, если у уравнения два корня.
+*
+*2. Инициализировать массив из целых чисел, описать функцию, принимающую на вход этот массив. Функция должна вернуть ноль, если в массиве нет нечётных чисел, в противном случае удвоить все нечётные числа в массиве и вернуть единицу. После выполнения функции, если массив был изменён, вывести все числа из массива на экран.
+*
+*3. Опишите функцию asShortPrint(unsigned* a, int len); которая принимает массив из тридцатидвухразрядных чисел (типа unsigned int), например, [0, 1, 2, 3, 4, 65541, 65542, 65543, 65544, 65545] и выводит его на экран шестнадцатиразрядными числами (типа unsigned short). Для массива в примере вывод будет выглядеть следующим образом [0, 0, 1, 0, 2, 0, 3, 0, 4, 0, 5, 1, 6, 1, 7, 1, 8, 1, 9, 1]. Решения такого рода широко применяются для расшифровывания данных из входящих потоков или логов цифровых систем в аппаратуре.
+*/
+#include <stdio.h>
+#include <math.h>
+
+// For all checking persons: variable definition for all problems. Don't comment the lines below
+#define ARRAY_LENGTH 10
+double x1, x2;
+int a, b, c;
+int diskr;
+char input;
+int digit_array[ARRAY_LENGTH];
+unsigned int unsigned_int_array[ARRAY_LENGTH];
+// don't comment the lines above this 
+
+int calcSquareEq(int a, int b, int c, double* x1, double* x2) {
+	if (a == 0)
+		return -1;
+	int d;
+	d = pow(b, 2) - 4 * a * c;
+	if (d < 0)
+		return -1;
+	else {
+		if (d > 0)
+			*x2 = (-b - sqrt(d)) / (2 * a);
+		*x1 = (-b + sqrt(d)) / (2 * a);
+		return (d > 0) ? 1 : 0;
+	}
+}
+
+void inputArray(int *arr, int len) {
+	int i = 0;
+	while (i < len) {
+		printf("Элемент № %2d: ", i);
+		scanf("%d", (arr + i++));
```

Comment:
Выражения вида `x + y++` обычно стараются разбивать, чтобы не возникало неопределенности порядка выполнения операций.
Тем более `i` - это переменная по которой организован цикл, очень удобно, когда её изменение происходит в явном виде (отдельноф строкой), а не в составе какого-то выражения.
```suggestion
		scanf("%d", arr + i));
		i++;
```